### PR TITLE
PageDirectory: need do gc twice in restore.

### PR DESCRIPTION
**Pull request by jiaqizho**
### What problem does this PR solve?

Issue Number: ref #4957 

Problem Summary:
- MVCC may not clean up deleted `entry` if that `entry` be refence by a deleted `ref`.
- For example: [p1, ref 2 -> 1, d 1, d 2]
- First time call `gcInMemEntries()`:
    - P1 entry 0 being_ref_count is not zero, won't be cleaned.
    - R2 entry 0 being_ref_count is zero, deref the P 1, make it being_ref_count is 0. Also R2 will be clean up.
- Second time call `gcInMemEntries()`:
    - P 1 being_ref_count is 0, it will be clean.
- If we can't clean up all of the deleted entries in `restore`. Then BlobStore may not restore success.

### What is changed and how it works?
- call `gcInMemEntries()` twice to clean up all of deleted entries.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp [hunk 1]
```diff
@@ -41,6 +41,22 @@ PageDirectoryPtr PageDirectoryFactory::create(String storage_name, FileProviderP
...
+    //  P 1
+    //  R 2 -> 1
+    //  D 1
+    //  D 2
+    // First time call gcInMemEntries():
+    //  - P 1 entry 0 being_ref_count is not zero, won't be cleaned.
+    //  - R 2 entry 0 being_ref_count is zero, deref the P 1, make it being_ref_count is 0. Also R2 will be clean up.
+    //
+    // Second time call gcInMemEntries():
+    //  - P 1 being_ref_count is 0, it will be clean.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp:57]: Can you add a test case for it?

Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp:57]: Can you add a test case for it?

Comment (jiaqizho) [reply dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp:57]: done

File: dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp [hunk 2]
```diff
@@ -42,6 +42,22 @@ PageDirectoryPtr PageDirectoryFactory::create(String storage_name, FileProviderP
...
+    dir->gcInMemEntries();
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp:59]: Also add it to `PageDirectoryFactory::createFromEdit`

Comment (jiaqizho) [reply dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp:59]: done

Comment (JaySon-Huang) [reply dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp:59]: I mean in these lines, we should also call gcInMemEntries twice
https://github.com/pingcap/tiflash/blob/62e7c14141c0a3390ac5a4454708126511382838/dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp#L87-L100

**Comment by JaySon-Huang** [comment]
Do not need this change because we have cleaned entries after decreasing the ref count in `VersionedPageEntries::derefAndClean`. 
https://github.com/pingcap/tiflash/blob/64a747a179d736475a66e02d926efbab5a133c32/dbms/src/Storages/Page/V3/PageDirectory.cpp#L599-L603

The test cases will be merged into https://github.com/pingcap/tiflash/pull/4987

**Pull request closed without merging**